github-for-jira.sd.yml

```diff
@@ -647,6 +647,75 @@ environmentOverrides:
             Period: 300
             ComparisonOperator: GreaterThanThreshold
             Statistic: Maximum
+
+      - type: sqs
+        name: backfill_dlq
+        alarms:
+          AlarmOnTooManyMessages:
+            MetricName: ApproximateNumberOfMessagesVisible
+            Namespace: AWS/SQS
+            Description: "Message count for backfill dead letter queue too high!"
+            Threshold: 5000 # number is initially high until we start bringing numbers down
```

Comment:
This is intentionally outrageously large at 5000.
We have massive amounts of messages in our DLQ currently and it averages 2000-3000 at any given time(in deployment dlq mainly). Something probably worth investigating, but so we don't get an alert every 20 mins 5000 was chosen as this seems to be the point where a spike will break out.

Open to options here tho!!

Also, once this is merged we will be monitoring the messaged that get tossed into the dlw bin and investigate how to bring them down.. long story short, this is not the final number we probably need to fix things before this alert becomes signal!

Reply:
I think investigating this more sounds like a solid idea. Can you create a ticket for disturbed/ops? (I hear @mboudreau is sorting the backlog and know he will love extra tickets to sort :badpokerface:)

Reply:
Think we need it to be 5000 for all queues? Just took a look at our DLQ metrics for the past year and it looks like the highest discovery ever reached was ~35 and backfill ~65. Push did get up above 6k but that was back in Dec last year. Makes sense for branch and deployments though (especially the later coz the metrics for that bad boy look like a rollercoaster). Not sure we need a one-size-fits all approach here.